CMakeLists.txt

```
@@ -40,11 +42,11 @@ if(CMAKE_Fortran_COMPILER_ID STREQUAL GNU)
   add_compile_options(-std=f2018)
 elseif(CMAKE_Fortran_COMPILER_ID MATCHES "^Intel")
   if(WIN32)
-    add_compile_options(/warn:declarations,general,usage,interfaces,unused)
-    add_compile_options(/stand:f18)
+    add_compile_options("$<$<COMPILE_LANGUAGE:Fortran>:/warn:declarations,general,usage,interfaces,unused>")
+    add_compile_options("$<$<COMPILE_LANGUAGE:Fortran>:/stand:f18>")
   else()
-    add_compile_options(-warn declarations,general,usage,interfaces,unused)
-    add_compile_options(-stand f18)
+    add_compile_options("$<$<COMPILE_LANGUAGE:Fortran>:-warn declarations,general,usage,interfaces,unused>")
+    add_compile_options("$<$<COMPILE_LANGUAGE:Fortran>:-stand f18>")
   endif()
```

```suggestion
  if(WIN32)
    set(fortran_flags /stand:f18 /warn:declarations,general,usage,interfaces,unused)
  else()
    set(fortran_flags -stand f18 -warn declarations,general,usage,interfaces,unused)
  endif()
  add_compile_options("$<$<COMPILE_LANGUAGE:Fortran>:${fortran_flags}>")
```
@awvwgk this should work fine (based on [this discussion](https://gitlab.kitware.com/cmake/cmake/-/issues/22230); It sounds also cleaner to me (but still suboptimal). What do you think?

---

Looks good to me.

---

Done